index.js

```diff
@@ -4,14 +4,38 @@
 
 var ensureString    = require('es5-ext/object/validate-stringifiable-value')
   , resolve         = require('path').resolve
+  , deferred        = require('deferred')
   , lstat           = require('fs2/lstat')
   , renameDirectory = require('./directory')
   , renameModule    = require('./module');
 
 module.exports = function (from, to) {
 	return lstat(resolve(ensureString(from)))(function (stats) {
-		if (stats.isDirectory()) return renameDirectory(from, to);
-		if (stats.isFile()) return renameModule(from, to);
+
+		// Dealing with case sensitive move by moving the original to a temporary location
+		// then move the temporary location to the destination.
+		// @Method: Temporary rename target to .movejstmp folder
+		// then rename it again to the original target name
+		var temporaryTo;
+		var promise;
+
+		if (from !== to && from.toLowerCase() === to.toLowerCase()) {
```

Comment:
Composition wise patches should be within `module.js` and `directory.js` modules, as direct usage of them should have same outcome

Interestingly you added tests, that test directly against this modules, so they can't pass :) However they pass on Travis, as it works with case sensitive system (where this change there has no effect).